Assets/Scripts/Client/BaseCornClient.cs

```diff
@@ -220,7 +222,7 @@ protected virtual void SetWorldOriginOffset(Vector3Int offset)
         public abstract bool UseItemOnOffHand();
         public abstract bool DoWindowAction(int windowId, int slotId, WindowActionType action);
         public abstract bool PlaceBlock(BlockLoc blockLoc, Direction blockFace, Hand hand = Hand.MainHand);
-        public abstract bool DigBlock(BlockLoc blockLoc, bool swingArms = true, bool lookAtBlock = true);
+        public abstract float DigBlock(BlockLoc blockLoc, Direction blockFace, DiggingStatus status = DiggingStatus.Started, bool swingArms = true, bool lookAtBlock = true);
```

Comment:
For this part of the code, do you have any good suggestions? I feel it makes the code messy.

Reply:
Actually, I could move the code elsewhere, but I don't want the scripts to reference each other.

Reply:
I think this part is fine, but maybe it's a good idea to put [tool digging effectiveness](https://github.com/DevBobcorn/CornCraft/pull/20/files#diff-e9f44e86b893676a15c40814f4bd69a3db533a94455ca51f114e4e7aebef2c0c) into item type definition or item type palette.

Reply:
Maybe we'll add some data about digging effectiveness to the item data files, and read this information into the item type definition.